src/Compilers/Core/Portable/PEWriter/MetadataWriter.cs

```diff
@@ -125,7 +124,7 @@ private bool IsMinimalDelta
         /// NetModules and EnC deltas don't have AssemblyDef record.
         /// We don't emit it for EnC deltas since assembly identity has to be preserved across generations (CLR/debugger get confused otherwise).
         /// </summary>
-        private bool EmitAssemblyDefinition => module.AsAssembly != null && !IsMinimalDelta;
+        private bool EmitAssemblyDefinition => module.OutputKind != OutputKind.NetModule && !IsMinimalDelta;
```

Comment:
Yes, it is correct. WinRT metadata files are assemblies. The only files that are not assemblies are NetModules.